numpy/random/mtrand.pyx

```diff
@@ -606,9 +606,16 @@ cdef class RandomState:
             high = low
             low = 0
 
-        key = np.dtype(dtype).name
+        dt = np.dtype(dtype)
+        key = dt.name
         if key not in _integers_types:
             raise TypeError('Unsupported dtype "%s" for randint' % key)
+        if dt.byteorder != '=' and dt.byteorder != '|':
+            import warnings
```

Comment:
I think this is fine to go to the top level. Can you add `# numpy 1.17.0, 2019-05-28` or similar as a comment to the deprecation.